### PR TITLE
Add CURL support for HTTPS connection

### PR DESCRIPTION
**Pull request by shrikant1407**
Add CURL support for HTTPS connection

Signed-off-by: Shrikant Temburwar <shrikant.temburwar@intel.com>

### CODE REVIEW

**Review by vikashmishra07 — APPROVED** [review]
Please update the readme for self signed certificate in respect to security consequence...

**Review by KiranSukhavasi — CHANGES_REQUESTED** [review]
hold on till FDO 1.1.0.3 release

File: cmake/cli_input.cmake [hunk 1]
```diff
@@ -184,32 +185,32 @@ set(CACHED_HTTPPROXY ${HTTPPROXY} CACHE STRING "Selected HTTPPROXY")
 message("Selected HTTPPROXY ${HTTPPROXY}")
 
 ###########################################
-# FOR PROXY_DISCOVERY
```
Comment (tajnisha) [inline cmake/cli_input.cmake]: Please retain PROXY_DISCOVERY as discussed.

File: lib/fdonet.c [hunk 2]
```diff
@@ -488,10 +490,18 @@ bool connect_to_manufacturer(fdo_ip_address_t *ip, uint16_t port,
 
 	if (is_mfg_proxy_defined()) {
 #if defined HTTPPROXY
+	if (ssl) {
+		if (!fdo_curl_proxy(&mfgproxy_ip, mfgproxy_port)) {
+			LOG(LOG_ERROR,
+		    "Invalid Proxy Connection info for Manufacturer server!\n");
```
Comment (tajnisha) [inline lib/fdonet.c:504]: Update the log message as "setting up proxy for MFG failed".

File: lib/fdoprotctx.c [hunk 3]
```diff
@@ -351,10 +352,16 @@ int fdo_prot_ctx_run(fdo_prot_ctx_t *prot_ctx)
...
+		size_t ctr = 0;
+		if (memset_s(tmp_buf, sizeof(tmp_buf), 0) != 0) {
+				LOG(LOG_ERROR, "Memset() failed!\n");
+				return false;
+			}
 
 		ret = fdo_con_recv_msg_header(prot_ctx->sock_hdl, &protver,
 					      (uint32_t *)&fdor->msg_type,
-					      &msglen, prot_ctx->ssl);
+					      &msglen, prot_ctx->ssl, tmp_buf, &ctr);
```
Comment (tajnisha) [inline lib/fdoprotctx.c:364]: Can tmp_buf be renamed as curl_buf and ctr as curl_buf_offset at all places ?

File: lib/fdonet.c (hunk 2 above)
Comment (tajnisha) [reply lib/fdonet.c:504]: Apply above for rv and owner function as well.

File: network/network_if_linux.c [hunk 4]
```diff
@@ -294,27 +514,23 @@ fdo_con_handle fdo_con_connect(fdo_ip_address_t *ip_addr, uint16_t port,
...
+			goto end;
+		}
+		*ssl = (void *)(intptr_t)(sock_hdl->sockfd);
+	} else {
+		sock_hdl->sockfd = socket(AF_INET, SOCK_STREAM, 0);
+		if (sock_hdl->sockfd < 0) {
 			goto end;
 		}
 
-		if (fdo_ssl_connect(*ssl)) {
```
[the remark this answers is not in the document]
Comment (shrikant1407) [reply network/network_if_linux.c:316]: It is still needed to pass unit tests

File: app/main.c [hunk 5]
```diff
@@ -279,7 +279,13 @@ int app_main(bool is_resale)
...
 #endif
 #if defined SELF_SIGNED_CERTS_SUPPORTED
-	if  (argc > 1 && (*argv[1] == '5' || *argv[2] == '5')) {
+	int strcmp_ss = 1;
+	int strcmp_ss2 = 1;
+
+	strcmp_s((char *)argv[1], DATA_CONTENT_SIZE, "'-ss'", &strcmp_ss);
+	strcmp_s((char *)argv[2], DATA_CONTENT_SIZE, "'-ss'", &strcmp_ss2);
+
+	if  (argc > 1 && (strcmp_ss || strcmp_ss2)) {
```
Comment (tajnisha) [inline app/main.c:288]: [out] | indicator | pointer to result indicator, greater than 0, equal to 0 or less than 0, if the string pointed to by dest is greater than, equal to or less than the string pointed to by src respectively.

Looks like the expected value of the 4th argument strcmp_ss is zero ?

**Review by tajnisha — DISMISSED** [review]

**Review by himanshu7196 — APPROVED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

**Pull request merged**